server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java

```diff
@@ -255,7 +259,9 @@ public void start(DiscoveryNode currentLeader, long term) {
         }
 
         private void sendHeartBeatToStore() {
-            sharedStore.writeHeartBeat(currentTerm, new HeartBeat(currentLeader, threadPool.absoluteTimeInMillis()));
+            if (register.readCurrentTerm() == currentTerm) {
```

Comment:
Could we have a `TODO` comment to indicate that we should become `CANDIDATE` if the term doesn't match? (Also if the write fails I guess)